src/nupic/types/Sdr.cpp

```diff
@@ -135,6 +135,24 @@ namespace sdr {
     SparseDistributedRepresentation::~SparseDistributedRepresentation()
         { deconstruct(); }
 
+
+    void SparseDistributedRepresentation::reshape(const vector<UInt> &dimensions) const {
+        // Make sure we have the data in a format which does not care about the
+        // dimensions, IE: dense or sparse but not coordinates
+        if( not dense_valid and not sparse_valid )
```

Comment:
why is both dense & sparse required here? getSparse() sets only sparse_valid

Reply:
Either dense or sparse is required here.
If neither dense nor sparse is available, then it gets the sparse data.

After it has gotten the data in a format which can be reshaped (sparse or dense) then it invalidates the coordinate format which can not be reshaped.